features/support/feature_tests/action/paypal.rb

```diff
@@ -0,0 +1,112 @@
+# coding: utf-8
+module FeatureTests
+  module Action
+    module Paypal
+      extend Capybara::DSL
+      extend RSpec::Matchers
+
+      module_function
+
+      def connect_marketplace_paypal
+        topbar = FeatureTests::Section::Topbar
+        paypal_preferences = FeatureTests::Section::MarketplacePaypalPreferences
+        admin_sidebar = FeatureTests::Section::AdminSidebar
+
+        # Connect Paypal for admin
+        topbar.navigate_to_admin
+        admin_sidebar.click_payments_link
+        paypal_preferences.connect_paypal_account
+
+        expect(page).to have_content("PayPal account connected")
+
+        # Save payment preferences
+        paypal_preferences.set_payment_preferences("2.0", "5", "1.0")
+        paypal_preferences.save_settings
+        expect(page).to have_content("Payment preferences updated")
+      end
+
+      def connect_seller_paypal
+        topbar = FeatureTests::Section::Topbar
+        settings_sidebar = FeatureTests::Section::UserSettingsSidebar
+        paypal_preferences = FeatureTests::Section::UserPaypalPreferences
+
+        # Connect Paypal for seller
+        topbar.open_user_menu
+        topbar.click_settings
+        settings_sidebar.click_payments_link
+        paypal_preferences.connect_paypal_account
+
+        expect(page).to have_content("PayPal account connected")
+
+        # Grant commission fee
+        paypal_preferences.grant_permission
+
+        expect(page).to have_content("Hooray, everything is set up!")
+      end
+
+      def request_listing(listing_title)
+        home = FeatureTests::Page::Home
+        listing = FeatureTests::Page::Listing
+        listing_book = FeatureTests::Page::ListingBook
+
+        home.click_listing(listing_title)
```

Comment:
All the actions seem to start with some `topbar` action, which is cool, because the topbar is available in all pages and thus the actions are available, where ever you are. So that made me think, should there be a topbar action also here? E.g. `topbar.click_logo`.

Or if there's a hard-dependency, that some action can be only performed when we already are in some page, then we could add `expect { ...you are in this and that page } `.

Does that make sense?